### PR TITLE
Remove galaxy-specifics from ansible repository list & detail tab

### PR DESCRIPTION
**Pull request by gerrod3**
I cleaning up the remaining galaxy bits from the ansible views and my PR was getting very big, so I am splitting them up and this is the first one.

### CODE REVIEW

File: src/components/ansible-repository-form.tsx [hunk 1]
```diff
@@ -103,20 +101,12 @@ export const AnsibleRepositoryForm = ({
...
-    hideFromSearch && pipeline === 'staging',
-  );
-
   const [remotes, setRemotes] = useState(null);
   const [remotesError, setRemotesError] = useState(null);
   const loadRemotes = (name?) => {
     setRemotesError(null);
     AnsibleRemoteAPI.list({ ...(name ? { name__icontains: name } : {}) })
-      .then(({ data }) => setRemotes(data.results))
+      .then(({ data }) => setRemotes(data.results.map((r) => ({...r, id: r.pulp_href}))))
```
Comment (himdel) [inline src/components/ansible-repository-form.tsx:109]: why the `id` field, I'm not seeing it used anywhere?

File: src/containers/ansible-repository/tab-details.tsx [hunk 2]
```diff
@@ -29,18 +26,6 @@ export const DetailsTab = ({ item }: TabProps) => {
           label: t`Retained version count`,
           value: item?.retain_repo_versions ?? t`All`,
         },
-        {
-          label: t`Distribution`,
-          value: <LazyDistributions repositoryHref={item?.pulp_href} />,
```
Comment (himdel) [inline src/containers/ansible-repository/tab-details.tsx:34]: Would we not want to show associated distributions & repo url?

**Comment by himdel** [comment]
Overall makes sense, thanks :+1: :)

File: src/components/ansible-repository-form.tsx (hunk 1 above)
Comment (gerrod3) [reply src/components/ansible-repository-form.tsx:109]: The value is used in `Typeahead` on line 199, they become the select options for the field and each of the options needs an id.

File: src/containers/ansible-repository/tab-details.tsx (hunk 2 above)
Comment (gerrod3) [reply src/containers/ansible-repository/tab-details.tsx:34]: There is technically already a distributions tab that will show associated distros, so I felt that was redundant. As for the repo url, that is actually a Galaxy thing. Although maybe I could be sold on its convenience, to encourage users to create 1-to-1 mappings between their repositories and distributions, to make operating Pulp simpler.

Comment (himdel) [reply src/containers/ansible-repository/tab-details.tsx:34]: Ooh, right, I remember now, base path of a first distribution associated with the repo, unless there's an identically named distribution :)

Yeaah, maybe let's keep it if it works and see if users find it useful? Maybe everybody uses distributions manually, but the repo=repo version=distribution abstraction definitely made things simpler :)

(I'm thinking the UI should make the simple flow easy, but still allow for more complicated ones?)

**Pull request merged**